### PR TITLE
[#362] Build doesn't fail with invalid test tags

### LINKED ISSUE #362: Build failes if no tests with tag matches found
This happens when we use test tag for example "core". We have such tests in robot tests but do not have in nose tests and at the stage of the building report job fails.

### PR DESCRIPTION
**Pull request by ablatov**
This closes #362 
1. If robot && nose test report HAVE test info = > successful build with boths reports 
2. If robot report HAVE test info = > successful build with ROBOT report.
3. If nose report HAVE test info = > successful build with NOSETEST report.
4. If NO robot && NO nose test report HAVE test info = > UNSTABLE build without reports

### CODE REVIEW

**Comment by ablatov** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test all

**Comment by ablatov** [comment]
@kirillmakhonin you were right about test run without tags. Only default tags executed. Fixing.

**Comment by ablatov** [comment]
bot, test all

**Comment by ablatov** [comment]
bot, test app

**Comment by ablatov** [comment]
bot, test all

**Comment by ablatov** [comment]
Failed tests: "Check if flower scale down works properly" was fixed in another PR.

**Comment by ablatov** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
@dsuslov some test tags existed only in one of tests group (robot, unit). `junit` parses results of tests execution that may be empty in there is no tags in test group.

Other possible solution: test file to be non-empty before invoking `junit`

**Comment by ablatov** [comment]
@dsuslov @kirillmakhonin 
1. If robot && nose test report HAVE test info = > successful build with boths reports 
2. If robot report HAVE test info = > successful build with ROBOT report.
3. If nose report HAVE test info = > successful build with NOSETEST report.
4. If NO robot && NO nose test report HAVE test info = > UNSTABLE build without reports

**Comment by ablatov** [comment]
bot, test app

File: deploy/legionPipeline.groovy [hunk 1]
```diff
@@ -206,19 +208,39 @@ def runRobotTests(tags="") {
...
+        junit 'tests/python/nosetests.xml'
+    }
+    else {
+        echo "No ''*.xml' files for generating nosetests report"
+    }
+    if (!(nose_report.toInteger() > 1 && robot_report.toInteger() > 0) && !tags) {
+        echo "All tests were run but no reports found. Marking build as UNSTABLE"
+        currentBuild.result = 'UNSTABLE'
+    }
+    if (!(nose_report.toInteger() > 1 && robot_report.toInteger() > 0) && tags) {
```
Comment (dsuslov) [inline deploy/legionPipeline.groovy]: @ablatov looks wrong.  correct - (!(nose_report.toInteger() > 1 || robot_report.toInteger() > 0) && tags)

**Comment by dsuslov** [comment]
bot, test app

**Pull request merged**